webpack/ForemanTasks/Components/TasksDashboard/TasksDashboardHelper.js

```diff
@@ -66,15 +66,31 @@ export const getQueryFromUrl = () => {
   return queryFromUriQuery(uriQuery);
 };
 
+export const updateURLQuery = ({ state, result, mode, time }) => {
+  const query = {
+    state,
+    result,
+    time_mode: mode === 'last' ? 'recent' : mode,
+    time_horizon: time,
+    page: 1,
+  };
+
+  const uri = new URI(window.location.href);
+  uri.setSearch(query);
+  window.history.pushState({ path: uri.toString() }, '', uri.toString());
+};
+
 export const resolveQuery = query => {
   const uriQuery = uriQueryFromQuery(query);
 
   const uri = new URI(window.location.href);
-  const { search } = uri.query(true);
+  const { search, per_page: perPage } = uri.query(true);
 
-  const data = { search, ...uriQuery, page: 1 };
+  const data = { search, ...uriQuery, page: 1, per_page: perPage };
   const { $, tfm } = window;
 
+  updateURLQuery(query);
```

Comment:
Why creating a separate function: `updateURLQuery` ?
instead of creating a new instance of `URI`, you can use the `uri` which was created here.

Reply:
wanted to have 2 small functions instead of 1 big function, reused the uri, thanks!